hugr-core/src/hugr/serialize.rs

```diff
@@ -48,24 +49,27 @@ enum Versioned<SerHugr = SerHugrLatest> {
 
 impl<T> Versioned<T> {
     pub fn new_latest(t: T) -> Self {
-        Self::V2(t)
+        Self::V3(t)
     }
 }
 
 impl<T: DeserializeOwned> Versioned<T> {
-    fn upgrade(mut self) -> Result<T, UpgradeError> {
+    fn upgrade(#[allow(unused_mut)] mut self) -> Result<T, UpgradeError> {
```

Comment:
The `mut` keyword is internal. We're still taking by value, so the interface doesn't change.
```suggestion
    fn upgrade(self) -> Result<T, UpgradeError> {
```